src/main/java/com/patina/codebloom/common/db/models/achievements/Achievement.java

```diff
@@ -0,0 +1,50 @@
+package com.patina.codebloom.common.db.models.achievements;
+
+import java.time.OffsetDateTime;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Builder
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@NoArgsConstructor
+@AllArgsConstructor
+public class Achievement {
+    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
+    private String id;
+
+    @NotNull
+    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
+    private String userId;
+
+    @NotNull
+    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
```

Comment:
this should have `nullable = true` since it can be null inside of the database.

```suggestion
    @Schema(requiredMode = Schema.RequiredMode.REQUIRED, nullable = true)
```